op/model.py

```diff
@@ -314,3 +326,15 @@ def config_get(self):
 
     def is_leader(self):
         return self._run('is-leader')
+
+    def pod_spec_set(self, spec, k8s_resources):
+        tmpdir = Path(tempfile.mkdtemp('-pod-spec-set'))
```

Comment:
Slightly better to have this right before the _try:_ line, and the two lines below inside it.